crates/stages/src/pipeline/mod.rs

```diff
@@ -86,6 +88,11 @@ pub struct Pipeline<DB: Database, U: SyncStateUpdater> {
     metrics: Metrics,
 }
 
+/// The future that returns the owned pipeline and the result of the pipeline run. See
+/// [Pipeline::run_as_fut].
+pub type PipelineFut<DB, U> =
+    Pin<Box<dyn Future<Output = (Pipeline<DB, U>, Result<ControlFlow, PipelineError>)> + Send>>;
```

Comment:
this looks a bit complex, I'd appreciate a new type or at least an alias for this